server/src/services/auth.service.ts

```diff
@@ -93,25 +97,21 @@ export class AuthService {
       accessing: false,
       owner: isOwner,
     };
-    return this.userModel.create(user);
+
+    const newUser = await this.userModel.create(user);
+    return { user: newUser, token: this.jwtService.sign(JSON.parse(JSON.stringify(newUser))) };
```

Comment:
Don't put all the user info into the token, just put what you need like roles and user id.
Tokens can be decrypted by anyone, so if shared, I will have there username, hash password and email

Reply:
Ok that certainly makes sense, I just made a payload type and made tweaks around that